src/platform/api/pythonApi.ts

```diff
@@ -487,19 +487,29 @@ export class InterpreterService implements IInterpreterService {
             return undefined;
         }
     }
-    private trackResolvedEnvironment(env?: ResolvedEnvironment) {
+    /**
+     * The Python Extension triggers changes to the Python environments.
+     * However internally we need to track changes to the environments as we wrap the Python extension API and the Python extension API only returns partial information.
+     * Some times what happens is
+     * - When we call get active interpreter we get some information from Python extension
+     * - We then come into this method and see the information has changed and we internally trigger a change event so other parts are aware of this
+     * - Next we call the Python extension API again, and the information is different yet again
+     * - We then trigger another change event
+     * - This goes on and on, basically the Python extension API returns different information for the same env.
+     *
+     * The argument `triggerChangeEvent` is more of a fail safe to ensure we don't end up in such infinite loops.
+     */
+    private trackResolvedEnvironment(env: ResolvedEnvironment | undefined, triggerChangeEvent: boolean) {
         if (env) {
             const resolved = pythonEnvToJupyterEnv(env);
-            let changed = false;
             if (
                 !this._interpreters.get(env.id) ||
                 !areObjectsWithUrisTheSame(resolved, this._interpreters.get(env.id)?.resolved)
             ) {
-                changed = true;
                 this._interpreters.set(env.id, { resolved });
-            }
-            if (changed) {
-                this.didChangeInterpreters.fire();
+                if (triggerChangeEvent) {
```

Comment:
should this still also check `changed`?

Reply:
it looks like we're now just firing this event for every interpreter found

Reply:
No need, if you check the code, previously `changed` was initialize din line 498, all i've done is moved this code into the same if block, which was effectively the same, & there by removing the unnecessary variable as well

Reply:
I.e. it will not fire everytime, only when there are changes (where previously we'd set `changed = true` in the same if condition